apps/remix-ide/src/app/editor/SourceHighlighters.js

```diff
@@ -28,13 +25,40 @@ class SourceHighlighters {
     }
   }
 
+  highlightAllFrom (from) {
+    try {
+      if (!this.highlighters[from]) return
+      const sourceHighlight = new SourceHighlighter()
```

Comment:
From what I see you don't need to re-instanciate each time a `sourceHighlight = new SourceHighlighter()` but only if it doesn't already exist

Reply:
they keep an ad for each marker and if I don't do it the new marker has different id

Reply:
> ad ?

Reply:
sorry. "an id"